Individual and Wildcard certificate (#933)
SUSE/doc-caasp: adoc/admin-monitoring-stack.adoc

====

=== Create DNS entries

In this example, we will use a master node with IP `10.86.4.158` in the case of NodePort service of the Ingress Controller.

[NOTE]
====
You should configure proper DNS names in any production environment.
These values are only for example purposes.
====

. Configure the DNS server
+
----
monitoring.example.com                      IN  A       10.86.4.158
prometheus.example.com                      IN  CNAME   monitoring.example.com
prometheus-alertmanager.example.com         IN  CNAME   monitoring.example.com
grafana.example.com                         IN  CNAME   monitoring.example.com
----

. Configure the management workstation `/etc/hosts` (optional)
+
----
10.86.4.158 prometheus.example.com prometheus-alertmanager.example.com grafana.example.com
----

==== TLS Certificate

You must configure your certificates for the components as secrets in the {kube} cluster.
Get certificates from your certificate authority.

. Individual certificate
+

> Single-name TLS certificate protects a single sub-domain, and it means each sub-domain owns its private key. From the security perspective, it is recommended to use a individual certificate. However, you have to manages the private key and the certificate rotation separately.

Reviewer: ```suggestion
Single-name TLS certificate protects a single sub-domain, and it means each sub-domain owns its private key. From the security perspective, it is recommended to use individual certificates. However, you have to manage the private key and the certificate rotation separately.
```

+
.Note Down Secret Names For Configuration
[IMPORTANT]
====

> Should you choose to secure each service with an individual certificate, you must repeat the step below for each component and adjust the name for the individual secret each time. Please note down the names of the secrets you have created.

Reviewer: ```suggestion
When you choose to secure each service with an individual certificate, you must repeat the step below for each component and adjust the name for the individual secret each time. Please note down the names of the secrets you have created.
```

In this example, the secret name is `monitoring-tls`.
====

. Wildcard certificate
+

> Wildcard TLS allows you to secure multiple sub-domains with one certificate and it means multiple sub-domains share the same private key. However, you could add more sub-domains without having to redeploy the certificate and more, saves the certificate costs.

Reviewer: ```suggestion
Wildcard TLS allows you to secure multiple sub-domains with one certificate and it means multiple sub-domains share the same private key. So, you could add more sub-domains without having to redeploy the certificate and more, saves the certificate costs.
```
